tools/profiles/debug.json

```diff
@@ -42,7 +42,7 @@
             "--diag_suppress=Pa050,Pa084,Pa093,Pa082", "-On", "-r", "-DMBED_DEBUG",
             "-DMBED_TRAP_ERRORS_ENABLED=1"],
         "asm": [],
-        "c": ["--vla"],
```

Comment:
why suddenly are not compatible ? This flag is only for C compiler ,thus should not cause any errors (we used to have it in common, and then I recall we faced an error with compatilibity , probably the same one you are seeing?) . See https://github.com/ARMmbed/mbed-os/pull/1985 why it was enabled.

This commit should be a separate pull request